src/editors/NonLinearDemandSupplyEditor.js

```diff
@@ -19,48 +20,72 @@ export default class NonLinearDemandSupplyEditor extends React.Component {
+                )}
                 <div className="form-row">
-                    <div className="col-sm-4">
-                        <label htmlFor="gLine1Slope">
-                            Orange line slope
-                        </label>
-                        <RangeEditor
-                            dataId="gLine1Slope"
-                            value={this.props.gLine1Slope}
-                            showOverrideCheckbox={true}
-                            overrideLabel='Vertical'
-                            overrideValue={999}
-                            handler={handleFormUpdate.bind(this)} />
-                    </div>
-                    <div className="col-sm-2">
-                        <div className="form-check">
-                            <label className="form-check-label">
-                                <input
-                                    id="gLine1SlopeEditable"
-                                    className="form-check-input"
-                                    type="checkbox"
-                                    onChange={handleFormUpdate.bind(this)}
-                                    checked={this.props.gLine1SlopeEditable} />
-                                Student editable
+                    {(this.props.isInstructor || this.props.gLine1SlopeEditable) && (
+                        <div className="col-sm-4">
+                            <label htmlFor="gLine1Slope">
+                                Orange line slope
                             </label>
+                            <RangeEditor
+                                dataId="gLine1Slope"
+                                value={this.props.gLine1Slope}
+                                showOverrideCheckbox={true}
+                                overrideLabel='Vertical'
+                                overrideValue={999}
+                                handler={handleFormUpdate.bind(this)} />
                         </div>
+                    )}
+                <div className="col-sm-2">
+                {this.props.isInstructor && (
+                    <div className="form-check">
+                        <label className="form-check-label">
+                            <input
+                                id="gLine1SlopeEditable"
+                                className="form-check-input"
+                                type="checkbox"
+                                onChange={handleFormUpdate.bind(this)}
+                                checked={this.props.gLine1SlopeEditable} />
+                            Student editable
+                        </label>
+                    </div>
+                )}
                     </div>
 
                 </div>
 
                 <div className="row">
-                    <div className="col">
-                        <label htmlFor="gCobbDouglasA">
-                            A
+                    {(this.props.isInstructor || this.props.gCobbDouglasAEditable) && (
```

Comment:
CobbDouglas properties referenced in the NonLinearDemandSupplyGraph. Is that as intended?